pkg/util/overridemanager/overridemanager_test.go

```diff
@@ -14,6 +19,223 @@ import (
+						Spec: policyv1alpha1.OverrideSpec{
+							ResourceSelectors: []policyv1alpha1.ResourceSelector{
+								{
+									APIVersion: "rbac.authorization.k8s.io/v1",
+									Kind:       "ClusterRole",
+									Name:       "test2",
+								},
+							},
+							OverrideRules: []policyv1alpha1.RuleWithCluster{
+								{
+									TargetCluster: &policyv1alpha1.ClusterAffinity{ClusterNames: []string{"test2"}},
+									Overriders: policyv1alpha1.Overriders{
+										LabelsOverrider: []policyv1alpha1.LabelAnnotationOverrider{
+											{
+												Operator: policyv1alpha1.OverriderOpAdd,
+												Value: map[string]string{
+													"testAddLabel": "testAddLabel",
+												},
+											},
+										},
+										AnnotationsOverrider: []policyv1alpha1.LabelAnnotationOverrider{
+											{
+												Operator: policyv1alpha1.OverriderOpAdd,
+												Value: map[string]string{
+													"testAddAnnotation": "testAddAnnotation",
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				).Build(),
+				EventRecorder: &record.FakeRecorder{},
+			},
+			args: args{
+				rawObj:      clusterRoleObj,
+				clusterName: "test2",
+			},
+			wantCOP: &AppliedOverrides{
+				AppliedItems: []OverridePolicyShadow{
+					{
+						PolicyName: "test2",
+						Overriders: policyv1alpha1.Overriders{
+							LabelsOverrider: []policyv1alpha1.LabelAnnotationOverrider{
+								{
+									Operator: policyv1alpha1.OverriderOpAdd,
+									Value: map[string]string{
+										"testAddLabel": "testAddLabel",
+									},
+								},
+							},
+							AnnotationsOverrider: []policyv1alpha1.LabelAnnotationOverrider{
+								{
+									Operator: policyv1alpha1.OverriderOpAdd,
+									Value: map[string]string{
+										"testAddAnnotation": "testAddAnnotation",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			wantOP:  nil,
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			o := &overrideManagerImpl{
+				Client:        tt.fields.Client,
+				EventRecorder: tt.fields.EventRecorder,
+			}
+			gotCOP, gotOP, err := o.ApplyOverridePolicies(tt.args.rawObj, tt.args.clusterName)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("ApplyOverridePolicies() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(gotCOP, tt.wantCOP) {
+				t.Errorf("ApplyOverridePolicies() gotCOP = %v, wantCOP %v", gotCOP, tt.wantCOP)
+			}
+			if !reflect.DeepEqual(gotOP, tt.wantOP) {
+				t.Errorf("ApplyOverridePolicies() gotOP = %v, wantOP %v", gotOP, tt.wantOP)
+			}
+			newLabels := tt.args.rawObj.GetLabels()
+			if _, ok := newLabels["testAddLabel"]; !ok {
+				t.Error("ApplyOverridePolicies() failed to add label")
+			}
+			newAnnotations := tt.args.rawObj.GetAnnotations()
+			if _, ok := newAnnotations["testAddAnnotation"]; !ok {
+				t.Error("ApplyOverridePolicies() failed to add annotation")
+			}
```

Comment:
I think that here can directly define an expected resource object and perform DeepEqual judgment on the resource object.

Reply:
ok